File: Frontend/lib/models/add_xp_response.dart
/// Modèle pour la réponse d'ajout d'XP
class AddXpResponse {
  final int xpAdded;
  final int totalXp;
  final int currentLevel;
  final String levelTitle;
  final int xpForNextLevel;
  final int xpProgressInCurrentLevel;
  final double progressPercentage;
  final bool leveledUp;
  final int? newLevel;
  final String message;

  AddXpResponse({
    required this.xpAdded,
    required this.totalXp,
    required this.currentLevel,
    required this.levelTitle,
    required this.xpForNextLevel,
    required this.xpProgressInCurrentLevel,
    required this.progressPercentage,
    required this.leveledUp,
    this.newLevel,
    required this.message,
  });

  factory AddXpResponse.fromJson(Map<String, dynamic> json) {
    return AddXpResponse(
      xpAdded: json['xpAdded'] ?? 0,
      totalXp: json['totalXp'] ?? 0,
      currentLevel: json['currentLevel'] ?? 1,
      levelTitle: json['levelTitle'] ?? 'Novice',
      xpForNextLevel: json['xpForNextLevel'] ?? 1000,
      xpProgressInCurrentLevel: json['xpProgressInCurrentLevel'] ?? 0,
      progressPercentage: (json['progressPercentage'] ?? 0).toDouble(),
      leveledUp: json['leveledUp'] ?? false,
      newLevel: json['newLevel'],
      message: json['message'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'xpAdded': xpAdded,
      'totalXp': totalXp,
      'currentLevel': currentLevel,
      'levelTitle': levelTitle,
      'xpForNextLevel': xpForNextLevel,
      'xpProgressInCurrentLevel': xpProgressInCurrentLevel,
      'progressPercentage': progressPercentage,
      'leveledUp': leveledUp,
      'newLevel': newLevel,
      'message': message,
    };
  }
}

File: Backend/src/main/java/com/example/dto/AddXpResponse.java
package com.example.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO pour la réponse après ajout d'XP
 * Retourne les nouvelles informations de progression
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AddXpResponse {
    
    private Integer xpAdded;
    private Integer totalXp;
    private Integer currentLevel;
    private String levelTitle;
    private Integer xpForNextLevel;
    private Integer xpProgressInCurrentLevel;
    private Double progressPercentage;
    private Boolean leveledUp; // Indique si l'utilisateur a gagné un niveau
    private Integer newLevel; // Le nouveau niveau (si leveledUp = true)
    private String message;
}